web/app/services/build-log-websocket.service.ts

```diff
@@ -0,0 +1,41 @@
+import {DOCUMENT} from '@angular/common';
+import {Inject, Injectable} from '@angular/core';
+import {Observable} from 'rxjs/Observable';
+import {Observer} from 'rxjs/Observer';
+import {Subject} from 'rxjs/Subject';
+
+export interface BuildLogMessageEvent extends MessageEvent {
+  data: string;
+}
+
+@Injectable()
+export class BuildLogWebsocketService {
+  private readonly API_ROOT: string;
+
+  constructor(@Inject(DOCUMENT) document: Document) {
+    this.API_ROOT = `ws://${document.location.host}/data`;
+  }
+
+  connect(projectId: string, buildId: string):
+      Observable<BuildLogMessageEvent> {
```

Comment:
Formatting is weird here, I wonder why the linter moved this to a new line

Reply:
I'm thinking it hit a line limit. I'm gonna have to set the linter to be inline with google styleguide at some point. 🤷‍♂️ for now